File: library/src/main/java/io/github/anderscheow/validator/rules/common/NotNullRule.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import io.github.anderscheow.validator.rules.BaseRule;

public class NotNullRule extends BaseRule {

    public NotNullRule() {
        super("Value must not be null");
    }

    public NotNullRule(@StringRes int errorRes) {
        super(errorRes);
    }

    public NotNullRule(@NonNull String errorMessage) {
        super(errorMessage);
    }

    @Override
    public boolean validate(Object value) {
        return value != null;
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/common/NotNullRule.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.Rule

class NotNullRule : Rule {

    constructor(@StringRes errorRes: Int) : super(errorRes)

    constructor(errorMessage: String) : super(errorMessage)

    override fun validate(value: String?): Boolean {
        return value != null
    }
}

fun notNull(@StringRes errorRes: Int): NotNullRule = NotNullRule(errorRes)

fun notNull(errorMessage: String): NotNullRule = NotNullRule(errorMessage)